x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/IndicesPermission.java

```diff
@@ -118,6 +118,10 @@ private StringMatcher indexMatcher(Collection<String> ordinaryIndices, Collectio
         return groups;
     }
 
+    public boolean isTotal() {
+        return Arrays.stream(groups).anyMatch(Group::isTotal);
```

Comment:
We can make this stricter to require the groups to have only one member, which is the case for superuser role and the API keys created by it.